### PR TITLE
server: Make per-peer inv relay handler a server method

### PR DESCRIPTION
**Pull request by jrick**
Future changes will relay inventory to specific peers.  This is a small amount of refactoring to prepare for that change.

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -2059,103 +2059,109 @@ func (s *server) maybeLogRecentlyAdvertisedNumEvicted() {
...
-		isBlockAnnouncement := iv.Type == wire.InvTypeBlock
-		if isBlockAnnouncement {
-			if sp.announcedBlock != nil && *sp.announcedBlock == iv.Hash {
-				sp.announcedBlock = nil
-				return
-			}
-			sp.announcedBlock = &iv.Hash
+		msgHeaders := wire.NewMsgHeaders()
+		if err := msgHeaders.AddBlockHeader(&blockHeader); err != nil {
+			peerLog.Errorf("Failed to add block"+
```
Comment (davecgh) [inline server.go:2096]: Same here.

File: server.go [hunk 2]
```diff
@@ -2059,103 +2059,109 @@ func (s *server) maybeLogRecentlyAdvertisedNumEvicted() {
...
+	}
 
-		// Ignore peers that do not have the required service flags.
-		if !hasServices(sp.Services(), msg.reqServices) {
+	// Generate and send a headers message instead of an inventory message
+	// for block announcements when the peer prefers headers.
+	if isBlockAnnouncement && sp.WantsHeaders() {
+		blockHeader, ok := msg.data.(wire.BlockHeader)
+		if !ok {
+			peerLog.Warnf("Underlying data for headers" +
```
Comment (davecgh) [inline server.go:2090]: This should fit without wrapping now due to the removed indent level.

Comment (jrick) [reply server.go:2090]: not quite if the goal is 80.

```
			peerLog.Warnf("Underlying data for inventory vector is not a transaction")
```

that's 98 chars wide with 8-space tabs and 86 with 4-space tabs.

Comment (davecgh) [reply server.go]: Oops, I commented on the wrong one.   I intended to comment on "Underlying data for headers is not a block header" and "Failed to add block header: %v".

Comment (davecgh) [reply server.go]: Oh, no I did comment on that one.  Hence the `blockHeader, ok` above.

Comment (jrick) [reply server.go]: errr that's a different warnf

Comment (davecgh) [reply server.go]: I guess it probably should be a `.Warn`  and not `Warnf` too now that I look at it since there are no params.

**Review by davecgh — APPROVED** [review]

**Pull request merged**